great_expectations/experimental/datasources/data_asset/data_connector/s3_data_connector.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, List, Optional
+
+from great_expectations.core._docs_decorators import public_api
+from great_expectations.core.batch_spec import PathBatchSpec, S3BatchSpec
+from great_expectations.experimental.datasources.data_asset.data_connector.file_path_data_connector import (
+    FilePathDataConnector,
+)
+from great_expectations.experimental.datasources.data_asset.data_connector.util import (
+    list_s3_keys,
+    sanitize_prefix_for_s3,
+)
+
+try:
+    import boto3
+except ImportError:
+    boto3 = None
+
+
+if TYPE_CHECKING:
+    from great_expectations.core.batch import BatchDefinition
+
+
+logger = logging.getLogger(__name__)
+
+
+@public_api
```

Comment:
We probably don't want to make this part of the public API yet.
Perhaps not at all since now this can simply be an implementation detail for each Datasource.

Reply:
@Kilo59 This was incorporated.  Thanks.